bin/zowe-configure-component.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/sh
+DEFAULT_TARGET_DIR=/opt/zowe/extensions
+
+if [[ -z ${ZOWE_ROOT_DIR} ]]
+then
+  export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+fi
+
+# Try and source the file utils if it exists
+if [[ -f "${ZOWE_ROOT_DIR}/bin/utils/file-utils.sh" ]]
+then
+    . ${ZOWE_ROOT_DIR}/bin/utils/file-utils.sh
+fi
+
+error_handler(){
+    print_error_message "$1"
+    exit 1
+}
+
+install_component(){
+
+    # Represents the current value of EXTERNAL_COMPONENT variable in instance.env
+    EXT_COMP_VAL=$(eval "grep '^EXTERNAL_COMPONENTS=' $INSTANCE_DIR/instance.env | cut -f2 -d= | cut -f1 -d#")
+
+    if [ "$EXT_COMP_VAL" = " " ]; then
+        sed -e "s/EXTERNAL_COMPONENTS=/EXTERNAL_COMPONENTS=$COMPONENT_NAME/" $INSTANCE_DIR/instance.env > $INSTANCE_DIR/instance.env.tmp
+        mv $INSTANCE_DIR/instance.env.tmp $INSTANCE_DIR/instance.env
+        log_message "Appending directory path containing component's life cycle scripts into the EXTERNAL_COMPONENTS variable in instance.env"
+    else
+        # Ensures that the bin directory of the component is included into the instance.env once (Avoids duplication if same component is installed twice)
+        if [[ $(grep 'EXTERNAL_COMPONENT' $INSTANCE_DIR/instance.env | grep $COMPONENT_NAME) = "" ]]; then
+            sed -e "s/EXTERNAL_COMPONENTS=/EXTERNAL_COMPONENTS=$COMPONENT_NAME,/" $INSTANCE_DIR/instance.env > $INSTANCE_DIR/instance.env.tmp
+            mv $INSTANCE_DIR/instance.env.tmp $INSTANCE_DIR/instance.env
+            log_message "Appending directory path containing component's life cycle scripts into the EXTERNAL_COMPONENTS variable in instance.env"
+        fi
+    fi
+
+    log_message "Zowe component has been installed."
+}
+
+install_desktop_plugin(){
+
+    log_message "Running $INSTANCE_DIR/bin/install-app.sh $COMPONENT_PATH"
+    # Uses install-app.sh in zowe-instance-dir to automatically set up the component onto zowe
+    $INSTANCE_DIR/bin/install-app.sh $COMPONENT_PATH
+
+    log_message "Zowe component has been installed."
+}
+
+configure_component(){
+
+    cd $COMPONENT_PATH
+
+    if [[ -f "$COMPONENT_PATH/manifest.yaml" ]]; then
+        MANIFEST_PATH="$COMPONENT_PATH/manifest.yaml"
+    elif [[ -f "$COMPONENT_PATH/manifest.yml" ]]; then
+        MANIFEST_PATH="$COMPONENT_PATH/manifest.yml"
+    elif [[ -f "$COMPONENT_PATH/manifest.json" ]]; then
+        MANIFEST_PATH="$COMPONENT_PATH/manifest.json"
+    fi
```

Comment:
What happens if `MANIFEST_PATH` is empty? The user could point to a wrong folder, or a legacy extension package without manifest.